### PR TITLE
Add `layout` property to `RadioCard`

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Add `@layout` property to `RadioCard`

### :hammer_and_wrench: Detailed description

Based on [feedback from our consumers](https://hashicorp.slack.com/archives/C7KTUHNUS/p1668039146943259), we extend the current layout behaviour for `RadioCard` items in a group to accommodate usecases where the cards have a fixed (or relatively fixed) width and wrap on multiple rows when exceeding parent's width. To achieve this we make the current behaviour default and name it `fluid` and we add a `fixed` layout where `@maxWidth` acts as a width and prevents flex items from growing, so they align naturally to left/start.

### :link: External links

[Slack thread](https://hashicorp.slack.com/archives/C7KTUHNUS/p1668039146943259)

***

### 👀 How to review

👉 Review by files changed

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/tests/dummy/app/templates/components/form/radio-card.hbs [hunk 1]
```diff
@@ -63,15 +63,36 @@
...
+      <p>Acceptable values:</p>
+      <ol>
+        <li class="default">fluid</li>
+        <li>fixed</li>
+      </ol>
       <p><em>Notice: by default the card will expand to fit the parent container. When used in a group the cards will
-          equally share the width to fit the available space. If a
+          equally share the width to fit the available space. If the
+          <code class="dummy-code">@layout</code>
+          parameter is set to fixed a
```
Comment (didoo) [inline packages/components/tests/dummy/app/templates/components/form/radio-card.hbs:76]:
```suggestion
          parameter is set to <code class="dummy-code">fixed</code> a
```

File: packages/components/app/styles/components/form/radio-card.scss [hunk 2]
```diff
@@ -15,9 +15,18 @@
...
+
+  // LAYOUT
+
+  .hds-form-radio-card--layout-fluid {
+    flex: 1;
+  }
+
+  .hds-form-radio-card--layout-fixed {
+    flex-basis: 100%;
+  }
```
Comment (didoo) [inline packages/components/app/styles/components/form/radio-card.scss:29]: Personally, when comes to `flex` I prefer explicit to implicit ;)

```suggestion
  .hds-form-radio-card--layout-fluid {
    flex: 1 0 0;
  }

  .hds-form-radio-card--layout-fixed {
    flex: 1 0 100%;
  }
```

Reason being, I always have to think of what the props are 😀

File: packages/components/addon/components/hds/form/radio-card/index.js [hunk 3]
```diff
@@ -64,6 +66,27 @@ export default class HdsFormRadioCardIndexComponent extends Component {
...
+   * Accepted values: fluid, fixed
+   *
+   * @param layout
+   * @type {string}
+   * @default 'fluid'
+   */
+  get layout() {
+    let { layout = DEFAULT_LAYOUT } = this.args;
+
+    assert(
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/components/addon/components/hds/form/radio-card/index.js:80]: Very good point – we have a similar assertion for the `Alert` so I used a similar copy for this one.

**Review by didoo — APPROVED** [review]
All good for me! Nice (and quick) job 🙂.

**Pull request merged**